src/Internal/Testnet/DistributeFunds.purs

```diff
@@ -1,252 +1,273 @@
-  , completeTxs: Nil
+type DistrFundsRoundResult wallet amount =
+  { sources :: List (SourceState wallet amount)
+  , deferredTargets :: List (wallet /\ amount)
   }
 
-parallelizedDistributionPlan
-  :: forall src target amount
-   . Map src (Array (Tx Unit target amount))
-  -> Array (Map src (Tx Unit target amount))
-parallelizedDistributionPlan _ = unsafeThrow "hello"
+data AssignUtxoResult wallet amount
+  = AssignUtxo_Unassigned
+  | AssignUtxo_Deferred
+  | AssignUtxo_AssignedToSource (SourceState wallet amount)
+
+derive instance Generic (AssignUtxoResult wallet amount) _
+derive instance (Eq wallet, Eq amount) => Eq (AssignUtxoResult wallet amount)
+
+instance (Show wallet, Show amount) => Show (AssignUtxoResult wallet amount) where
+  show = genericShow
 
 makeDistributionPlan
-  :: forall src target amount
-   . Ord src
-  => Ord amount
-  => Ord target
-  => Ring amount
-  => Map src amount
-  -> Map target (Array amount)
-  -> { maxCoinPerTx :: amount
-     , maxTargetUtxosPerTx :: Int
-     }
-  -> Either
-       { err :: String
-       , acc :: List (SourceState src target amount)
-       }
-       (Map src (Array (Tx Unit target amount)))
-makeDistributionPlan sources targets thresholds = do
-  let
-    targetsUtxosAsc :: List { key :: target, amount :: amount }
-    targetsUtxosAsc = List.sortBy (flip compare)
-      $ Map.toUnfoldable targets
-      >>= \(key /\ utxos) ->
-        { key, amount: _ } <$> List.fromFoldable utxos
-
-    assigned :: Either _ (Map src (Array (Tx Unit target amount)))
-    assigned = do
-      sourcesTxs <- foldM
-        (flip $ assignUtxo thresholds)
-        ( initialSourceState <<< uncurry { key: _, initialFunds: _ } <$>
-            Map.toUnfoldable sources
-        )
-        targetsUtxosAsc
-      let
-        finish src =
-          src
-            # (_tx .~ emptyTx zero)
-            # (_completeTxs %~ Cons (src ^. _tx))
-        sourceToTxs = Map.fromFoldable
-          $ Tuple
-          <<< view _source
-          <*> Array.fromFoldable
-          <<< view _completeTxs
-          <<< finish
-          <$> sourcesTxs
-      pure sourceToTxs
-  assigned
-
-assignUtxo
-  :: forall target src amount
+  :: forall wallet amount
    . Ord amount
   => Ring amount
-  => { maxCoinPerTx :: amount
-     , maxTargetUtxosPerTx :: Int
-     }
-  -> { amount :: amount, key :: target }
-  -> List (SourceState src target amount)
-  -> Either
-       { err :: String
-       , acc :: List (SourceState src target amount)
-       }
-       (List (SourceState src target amount))
-assignUtxo _ _ Nil = Left
-  { err: "Ran out of sources", acc: Nil }
-assignUtxo thresholds utxo acc@(Cons source sources)
-  | 0 >= thresholds.maxTargetUtxosPerTx =
-      Left { err: "maxTargetUtxosPerTx must be greater than 1", acc }
-  | utxo.amount >= thresholds.maxCoinPerTx =
-      Left
-        { err: "UTxO required amount is higher than the maxCoinPerTx threshold"
-        , acc
+  => DistrFundsParams wallet amount
```

Comment:
A written description would help

Reply:
Addressed in https://github.com/Plutonomicon/cardano-transaction-lib/pull/1644/commits/b881710f7c5cbda8533a663d9caf5af252fc6329